source/common/common/assert.h

```diff
@@ -74,6 +74,24 @@ void invokeDebugAssertionFailureRecordAction_ForAssertMacroUseOnly();
  */
 #define SECURITY_ASSERT(X, DETAILS) _ASSERT_IMPL(X, #X, abort(), DETAILS)
 
+#if defined(ENVOY_DEBUG_KNOWN_ISSUES)
+/**
+ * Assert macro for an as-yet unidentified issue. Even with ASSERTS compiled in, it will be
+ * excluded, unless ENVOY_DEBUG_KNOWN_ISSUES is defined. It represents a condition that
+ * should always pass but that sometimes fails for an unknown reason. The macro allows it to
+ * be temporarily compiled out while the failure is triaged and investigated.
+ */
+#define KNOWN_ISSUE_ASSERT(X, DETAILS) _ASSERT_IMPL(X, #X, abort() DETAILS)
```

Comment:
Do you need release and non-release variants of this? Also can't you just define in terms of ASSERT and RELEASE_ASSERT?